internal/host/validator.go

```diff
@@ -1526,3 +1526,24 @@ func (v *validator) printVSphereUUIDEnabled(_ *validationContext, status Validat
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) compatibleAgent(c *validationContext) ValidationStatus {
+	if !v.hwValidatorCfg.EnableUpgradeAgent {
+		return ValidationSuccess
+	}
+	if common.IsAgentCompatible(v.hwValidatorCfg.AgentDockerImage, c.host.DiscoveryAgentVersion) {
+		return ValidationSuccess
+	}
+	return ValidationFailure
+}
+
+func (v *validator) printCompatibleAgent(c *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
```

Comment:
I think we should have a different message in case `c.hwValidatorCfg.EnableUpgradeAgent` is true

Reply:
I guess you meant when it is `false`. Do you mean something like this?

```diff
diff --git a/internal/host/validator.go b/internal/host/validator.go
index 27c0248f..d02fb915 100644
--- a/internal/host/validator.go
+++ b/internal/host/validator.go
@@ -1540,7 +1540,10 @@ func (v *validator) compatibleAgent(c *validationContext) ValidationStatus {
 func (v *validator) printCompatibleAgent(c *validationContext, status ValidationStatus) string {
        switch status {
        case ValidationSuccess:
-               return "Host agent is compatible with the service"
+               if v.hwValidatorCfg.EnableUpgradeAgent {
+                       return "Host agent is compatible with the service"
+               }
+               return "Host agent compatibility checking is disabled"
        case ValidationFailure:
                return "This host's agent is in the process of being upgraded to a newer version. " +
                        "This might take a few minutes"
```

Reply:
Yes

Reply:
Done.